src/navier_stokes/INSStaggeredHierarchyIntegrator.cpp

```diff
@@ -584,7 +584,7 @@ INSStaggeredHierarchyIntegrator::INSStaggeredHierarchyIntegrator(std::string obj
     d_Q_var = INSHierarchyIntegrator::d_Q_var;
     d_N_old_var = new SideVariable<NDIM, double>(d_object_name + "::N_old");
 
-    d_U_cc_var = new CellVariable<NDIM, double>(d_object_name + "::U_cc", NDIM);
+    d_U_nc_var = new NodeVariable<NDIM, double>(d_object_name + "::U_nc", NDIM);
```

Comment:
I think that you need to do:
```suggestion
    d_U_nc_var = new NodeVariable<NDIM, double>(d_object_name + "::U_nc", NDIM, /*fine_boundary_represents_var*/ false);
```
and then _refine_ from coarse to fine levels to get continuity at coarse-fine interfaces.

Reply:
The syncronization would be:
1. _coarsen_ data from fine levels to coarse levels (using `OuternodeVariable`/`OuternodeData` for efficiency).
2. _refine_ data from coarse levels to fine levels (using `LINEAR_REFINE`) to ensure everything matches along the coarse-fine interface.

Reply:
(All of the synchronization probably belongs in `HierarchyMathOps::interp()`. It would be great if the implementation was flexible enough to handle both `fine_boundary_represents_var==true` and `fine_boundary_represents_var==false`.)

Reply:
> (All of the synchronization probably belongs in `HierarchyMathOps::interp()`. It would be great if the implementation was flexible enough to handle both `fine_boundary_represents_var==true` and `fine_boundary_represents_var==false`.)

Basically, if `fine_boundary_represents_var==true`, then synching the values just requires step 1 (coarsening) and not step 2 (refining) --- step 2 will not change any grid values.

Reply:
One of the points of writing nodal data is to get a continuous representation into VisIt, so that things like isocontours are continuous.

Reply:
Notice also that the synchronization is almost the same as what is already done for side-centered data in `HierarchyMathOps` --- it uses `OutersideData` to synch up values on coarse-fine interfaces.

Reply:
I think that `CONSTANT_COARSEN` already provides the correct injection-based coarsen operator (see: https://ibamr.github.io/IBAMR-docs/samrai/2.4.4/html/package_patchdata_operators.html).

Reply:
Oh yeah, that works!

![visit0004](https://user-images.githubusercontent.com/3917035/194164947-8125b6af-1c33-42b0-8125-144eaf1e7c26.png)

I think the very small jumps are due to VisIt itself. I'll write some test code to see if I can verify that.

Reply:
Despite my claims to the contrary elsewhere, it is probably OK just to coarsen and refine a linear nodal field and verify that nothing changes. You just want to make sure that `fine_boundary_represents_var==false` in the `NodeVariable` constructor.

Reply:
(Note that if you coarsen directly from the `NodeData` it may not do the right thing at coarse-fine interfaces if `fine_boundary_represents_var==false` -- although I am not sure!)

Reply:
I spent a few hours this afternoon looking at this with a reduced test case and I think what we need (computing boundary values so that the solution is H1-conforming) is missing from SAMRAI - it only seems to schedule the priority of operators with this information but I cannot find any case in which it actually uses this information to go back and 'fix' the nodes on patch boundaries. I'll put something together that does this.

Reply:
> In particular, `CONSTANT_COARSEN` makes fine-grid nodes match the coarse-grid ones, but it does not go on to fill in the hanging nodes (which it really should!)

Wait a minute, `CONSTANT_COARSEN` changes the _fine_ data?

Reply:
I was kinda tired when I did this work on Friday - it at least synchronized them. I'll check!